source/core/NumericString.cpp

```diff
@@ -559,33 +582,49 @@ void Format(SubString *format, Int32 count, StaticTypeAndData arguments[], Strin
                             if (precision < 0) {
                                 precision = 0;
                                 truncateSignificant = exponent + 1 - fOptions.Significant;
-                                // need truncate the integer part of the float because the sprintf doesnt do this for us.
+                                // need truncate the inxteger part of the float because the sprintf doesnt do this for us.
                             }
                         }
-                        char asciiReplacementString[2*kTempCStringLength];
-                        Int32 sizeOfNumericString = -1;
                         char formatCode[10];
+                        skipPrev = Int32(strlen(replacementString));
                         if (precision >= 0) {
                             sprintf(formatCode, "%%.*%sf", fOptions.NumericLength);
                             // formatCode : %.*hf
-                            sizeOfNumericString = snprintf(asciiReplacementString, sizeof(asciiReplacementString), formatCode, precision,tempDouble);
+                            sizeOfNumericString = snprintf(replacementString+skipPrev, sizeof(replacementString)-skipPrev, formatCode, precision,tempDouble);
                         } else {
                             sprintf(formatCode, "%%%sf", fOptions.NumericLength);
                             // formatCode: %hf
-                            sizeOfNumericString = snprintf(asciiReplacementString, sizeof(asciiReplacementString), formatCode, tempDouble);
+                            sizeOfNumericString = snprintf(replacementString+skipPrev, sizeof(replacementString)-skipPrev, formatCode, tempDouble);
                         }
-                        Int32 intDigits = (exponent >= 0)? (exponent): 0 ;
-                        RefactorLabviewNumeric(&fOptions, asciiReplacementString, &sizeOfNumericString, intDigits, truncateSignificant);
-                        UpdateNumericStringWithDecimalSeparator(fOptions, asciiReplacementString, sizeOfNumericString);
-                        buffer->Append(sizeOfNumericString, (Utf8Char*)asciiReplacementString);
+                        sizeOfNumericString += skipPrev;
+                        if (exponent > 0)
+                            intDigits = exponent;
+                        // further processing done by shared code which calls RefactorLVNumeric below switch
                         argumentIndex++;
                     }
                     break;
                     case 'e': case 'E':
                     {
                         TypeRef argType = arguments[argumentIndex]._paramType;
-                        Double tempDouble = ReadDoubleFromMemory(argType,  arguments[argumentIndex]._pData);
+                        void *pData = arguments[argumentIndex]._pData;
```

Comment:
I actually went ahead and did this by moving it before the switch instead of making an extra function.
I didn't do this before because it requires extra checks for format code and that the argument is safe to access, but it's probably still worth it, so I did it.
During the process, I noticed an issue with error handling and fixed that too.